partiql-planner/src/main/kotlin/org/partiql/planner/internal/PartiQLHeader.kt

```diff
@@ -182,15 +184,24 @@ internal object PartiQLHeader : Header() {
         unary("neg", t, t)
     }
 
-    private fun eq(): List<FunctionSignature.Scalar> = types.all.map { t ->
+    private fun eq(): List<FunctionSignature.Scalar> = types.all.filterNot { it == ANY || it == MISSING }.map { t ->
```

Comment:
There's a divergence here between the modeling of functions between this PR and the `v1` branch. E.g.
- For `eq`, the `v1` branch includes all the types and they have the same `isNullable, `isMissingCall`, etc. configurations https://github.com/partiql/partiql-lang-kotlin/blob/2879f3a0a1d3197cc00fda6f1613490f88d19353/partiql-spi/src/test/kotlin/org/partiql/spi/connector/sql/SqlHeader.kt#L233-L241
- For the logical ops, the `v1` branch uses the defaults for `isMissable = true` and `isMissingCall = true`, whereas this PR defines `isMissable = false` and `isMissingCall = false`

I'm curious the reasoning for the modeling differences.